virtdbg/vmx/vmm.c

```diff
@@ -0,0 +1,290 @@
+#include <vmx/vmm.h>
+#include <mm/pmm.h>
+#include <arch/intrin.h>
+#include <stdint.h>
+#include <vmx/ept.h>
+#include <stddef.h>
+#include <util/except.h>
+#include <virtdbg.h>
+
+extern void vm_resume(guest_state_t *t);
+extern ept_entry_t* g_root_pa;
+
+//enable vmx operation
+err_t vmxon() {
+    err_t err = NO_ERROR;
+    uint32_t* vmxon_region = pallocz_aligned(0x1000, 0x1000);
+
+    size_t control = __rdmsr(0x3a);
+    if((control & (0x1 | 0x4)) != (0x1 | 0x04)) {
```

Comment:
define these bits nicely

Reply:
same for any other constants that have real meaning in this page, if it is a bitfield you could also define a struct like the cr0/cr4 ones.